### ISSUE TITLE
Touch and tablet events in X11 backend

### ISSUE THREAD

**Issue by i509VCB** [Smithay/smithay#405]
For parity with the Winit backend, we probably want to implement touch event support for the X11 backend.

Regarding implementation, we can optionally depend on the xinput extension to receive touch events from the X server. I believe tablet events can also be received from xinput as well.

**Comment by psychon** [comment]
Random thought on how this could be approached when one has a suitable device: Run `xprop`, `xinput --test` and `xinput --test-xi2` in three terminals and check what they say when some input events come in.

**Comment by i509VCB** [comment]
> Random thought on how this could be approached when one has a suitable device: Run `xprop`, `xinput --test` and `xinput --test-xi2` in three terminals and check what they say when some input events come in.

I have a touchscreen laptop and several drawing tablets I can take off my brother, so no worry about device availability for me at least.

**Comment by PolyMeilex** [comment]
Yeah I have a wacom tablet too, so we should be fine on that front.
